cordapp-cpk/src/test/kotlin/net/corda/plugins/cpb/CpbTest.kt

```diff
@@ -0,0 +1,131 @@
+package net.corda.plugins.cpb
+
+import net.corda.plugins.cpk.GradleProject
+import net.corda.plugins.cpk.cordaApiVersion
+import net.corda.plugins.cpk.deleteAll
+import net.corda.plugins.cpk.expectedCordappContractVersion
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestReporter
+import org.junit.jupiter.api.io.TempDir
+import java.io.InputStream
+import java.nio.file.Files
+import java.nio.file.Path
+import java.security.DigestInputStream
+import java.security.MessageDigest
+import java.security.cert.X509Certificate
+import java.util.TreeMap
+import java.util.jar.JarInputStream
+import javax.security.auth.x500.X500Principal
+import kotlin.streams.asSequence
+
+class CpbTest {
+    companion object {
+        private const val cordappVersion = "1.2.1"
+        private const val hostVersion = "2.0.1"
+        private val cordaDevCertPrincipal = X500Principal("CN=Corda Dev Code Signer, OU=R3, O=Corda, L=London, C=GB")
+
+
+        private lateinit var externalProject: GradleProject
+        private lateinit var testProject: GradleProject
+        private lateinit var testDir: Path
+        private lateinit var mavenRepoDir : Path
+        private lateinit var externalCordappProjectDir : Path
+        private lateinit var cpbProjectDir : Path
+
```

Comment:
I think all of these `Path` properties can become local variables now. In fact, `@TempDir tmpDir` could probably be replaced by `@TempDir testDir`.

Reply:
but I use them in multiple methods.. how can they become local variable? I assign them in `@BeforeAll` and use them in `@Test`

Reply:
> but I use them in multiple methods.. how can they become local variable? I assign them in `@BeforeAll` and use them in `@Test`

But that's the point - you _don't_ use them in multiple methods any more. At least, not once the `@AfterAll` method becomes
```kotlin
@AfterAll
fun tearDown() = externalProject.delete()
```
BTW, does JUnit execute that `@AfterAll` method if it hasn't also been annotated with `@JvmStatic`?